src/AMSWorkflow/ams/rmq.py

```diff
@@ -55,20 +56,23 @@ def endianness(self) -> str:
         """
         return "="
 
-    def encode(self, num_elem: int, input_dim: int, output_dim: int, dtype_byte: int = 4) -> bytes:
+    def encode(self, num_elem: int, domain_name: str, input_dim: int, output_dim: int, dtype_byte: int = 4) -> bytes:
         """
         For debugging and testing purposes, this function encode a message identical to what AMS would send
         """
-        header_format = self.endianness() + self.header_format()
+        header_format = self.ams_endianness() + self.ams_header_format()
         hsize = struct.calcsize(header_format)
         assert dtype_byte in [4, 8]
         dt = "f" if dtype_byte == 4 else "d"
         mpi_rank = 0
         data = np.random.rand(num_elem * (input_dim + output_dim))
-        header_content = (hsize, dtype_byte, mpi_rank, data.size, input_dim, output_dim)
+        domain_name_size = len(domain_name)
+        domain_name = bytes(domain_name, "utf-8")
+        padding = 0
+        header_content = (hsize, dtype_byte, mpi_rank, domain_name_size, data.size, input_dim, output_dim, padding)
         # float or double
-        msg_format = f"{header_format}{data.size}{dt}"
-        return struct.pack(msg_format, *header_content, *data)
+        msg_format = f"{header_format}{domain_name_size}s{data.size}{dt}"
```

Comment:
What is the s between the domain_name_size and data.size?

Reply:
It indicates a string here (cf [Python struct format](https://docs.python.org/3/library/struct.html#format-characters))

Reply:
Thnx